index/scorch/introducer.go

```diff
@@ -213,6 +216,10 @@ func (s *Scorch) introduceSegment(next *segmentIntroduction) error {
 	if next.persisted != nil {
 		s.rootPersisted = append(s.rootPersisted, next.persisted)
 	}
+	if next.callbacks != nil {
+		s.callbacks = append(s.callbacks, next.callbacks...)
+	}
```

Comment:
Now it looks like lines 219-thru-221 need to be removed, as line 222 takes care of things -- otherwise the callback is appended twice()?

Reply:
you are right, this was a bad merge ...
Fixed